### PR TITLE
NO_JIRA: Added service tests for paths not covered by intergration tests

### PR DESCRIPTION
**Pull request by bills7281**
1. Added missing service tests.
2. Corrected 'nulls in the wild' manifesting in UI by giving default empty Strings for all constituent String fields.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      given(communityApiClient.getConvictions(anyString()))
+        .willReturn(Mono.empty())
+
+      val response = caseSummaryOverviewService.getOverview(crn)
+
+      val personalDetails = response.personDetails!!
+      val offencesShouldBeEmpty = response.offences
+
+      assertThat(personalDetails.crn).isEqualTo(crn)
+      assertThat(personalDetails.age).isEqualTo(39)
```
Comment (jackmaloney) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt:76]: This age will break when the person becomes a year older. I've tried to handle this in similar way here: https://github.com/ministryofjustice/make-recall-decision-api/pull/62/files#diff-68e948597daf542f1164ac2e0971005beb2490eae21919aa41be449f5e38e48cR142

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      assertThat(personalDetails.gender).isEqualTo("Male")
+      assertThat(personalDetails.dateOfBirth).isEqualTo(LocalDate.parse("1982-10-24"))
+      assertThat(personalDetails.name).isEqualTo("John Smith")
+      assertThat(offencesShouldBeEmpty).isNull()
+      then(communityApiClient).should().getAllOffenderDetails(crn)
+    }
+  }
```
Comment (jackmaloney) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt:83]: Think you're missing a test to show the happy path for offences when they get returned.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/OffenderSearchServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  @Mock
+  private lateinit var offenderSearchApiClient: OffenderSearchApiClient
+
+  @BeforeEach
+  fun setup() {
+    offenderSearch = OffenderSearchService(offenderSearchApiClient)
+  }
+
+  @Test
```
Comment (jackmaloney) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/OffenderSearchServiceTest.kt:31]: Can we add a happy path when search returns results

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PersonalDetailServiceTest.kt [hunk 4]
```diff
@@ -61,74 +62,88 @@ class PersonalDetailServiceTest {
...
-      // TODO assert details
+      val response = personDetailsService.getPersonDetails(crn)
+
+      val personalDetails = response.personDetails!!
+      val currentAddress = response.currentAddress!!
+      val riskFlagsShouldBeEmpty = response.risk!!.flags
+      val offenderManager = response.offenderManager!!
 
+      assertThat(personalDetails.crn).isEqualTo(crn)
+      assertThat(personalDetails.age).isEqualTo(39)
```
Comment (jackmaloney) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PersonalDetailServiceTest.kt:78]: Same comment here about age

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PersonalDetailServiceTest.kt [hunk 5]
```diff
@@ -61,74 +62,88 @@ class PersonalDetailServiceTest {
...
-          Address(
-            town = "Sheffield",
-            county = "South Yorkshire",
-            buildingName = "HMPPS Digital Studio",
-            district = "Sheffield City Centre",
-            status = AddressStatus(code = "ABC123", description = "Not Main"),
-            postcode = "S3 7BS",
-            addressNumber = "33"
-          )
+  private val allOffenderDetailsResponse = AllOffenderDetailsResponse(
```
Comment (jackmaloney) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PersonalDetailServiceTest.kt:96]: Probably need to add a happy path test for when 'AllOffenderDetails' and 'Registrations' are both retrieved.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt (hunk 2 above)
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt:83]: That is covered in the integration tests. Shall I add one here too?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/OffenderSearchServiceTest.kt (hunk 3 above)
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/OffenderSearchServiceTest.kt:34]: Done

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt (hunk 1 above)
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt]: Done

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PersonalDetailServiceTest.kt (hunk 4 above)
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PersonalDetailServiceTest.kt]: Done

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PersonalDetailServiceTest.kt (hunk 5 above)
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PersonalDetailServiceTest.kt]: Done

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**